mep/people/views.py

```diff
@@ -7,44 +7,106 @@
 from django.urls import reverse
 from django.utils.html import format_html
 from django.utils.safestring import mark_safe
-from django.views.generic import DetailView, ListView
-from django.views.generic.edit import FormView
+from django.utils.timezone import now
+from django.views.generic import ListView, DetailView
+from django.views.generic.edit import FormView, FormMixin
 from parasolr.django import SolrQuerySet
 
 from mep.accounts.models import Event
 from mep.common.utils import alpha_pagelabels
 from mep.common.views import LabeledPagesMixin
-from mep.people.forms import PersonMergeForm
+from mep.people.forms import PersonMergeForm, MemberSearchForm
 from mep.people.geonames import GeoNamesAPI
 from mep.people.models import Country, Location, Person
 
 
-class MembersList(LabeledPagesMixin, ListView):
+class MembersList(LabeledPagesMixin, ListView, FormMixin):
     '''List page for searching and browsing library members.'''
     model = Person
     template_name = 'people/member_list.html'
     paginate_by = 100
     context_object_name = 'members'
 
+    form_class = MemberSearchForm
+    # cached form instance for current request
+    _form = None
+    #: initial form values
+    initial = {
+        'sort': 'name'
+    }
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        # use GET instead of default POST/PUT for form data
+        form_data = self.request.GET.copy()
+
+        # always use relevance sort for keyword search;
+        # otherwise use default (sort by name)
+        if 'query' in form_data:
+            form_data['sort'] = 'relevance'
+        else:
+            form_data['sort'] = self.initial['sort']
+
+        # use initial values as defaults
+        for key, val in self.initial.items():
+            form_data.setdefault(key, val)
+
+        kwargs['data'] = form_data
+        return kwargs
+
+    def get_form(self, *args, **kwargs):
+        # initialize the form, caching on current instance
+        if not self._form:
+            self._form = super().get_form(*args, **kwargs)
+        return self._form
+
+    #: name query alias field syntax
+    search_name_query = '{!type=edismax qf=$name_qf pf=$name_pf v=$name_query}'
```

Comment:
Just realized, we shouldn't need `edismax` specified here, because we do that at the `solrconfig` level for the handler.

Reply:
This syntax I've found in documentation and examples for querying on a field alias always includes that, but I just did a quick test and you're right.